web/services/AI/ai_model.py

```
@@ -0,0 +1,173 @@
+
+    # Generate response
+    try:
+        response = model.generate_content(user_input)
+
+        # Try to extract JSON from the response
+        json_match = re.search(r"\{.*\}", response.text, re.DOTALL)
+        if json_match:
+            json_str = json_match.group(0)
+            response_object = json.loads(json_str)
+        else:
+            raise ValueError("No JSON object found in response")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)